diff -up ./crypto/CryptoAuth.c.warnings ./crypto/CryptoAuth.c
--- ./crypto/CryptoAuth.c.warnings	2017-02-21 17:56:22.000000000 -0500
+++ ./crypto/CryptoAuth.c	2017-02-24 12:36:55.762819179 -0500
@@ -85,6 +85,8 @@ static inline void getSharedSecret(uint8
                                    uint8_t passwordHash[32],
                                    struct Log* logger)
 {
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-result"
     if (passwordHash == NULL) {
         crypto_box_curve25519xsalsa20poly1305_beforenm(outputSecret, herPublicKey, myPrivateKey);
     } else {
@@ -100,6 +102,7 @@ static inline void getSharedSecret(uint8
         Bits_memcpy(buff.components.passwd, passwordHash, 32);
         crypto_hash_sha256(outputSecret, buff.bytes, 64);
     }
+#pragma GCC diagnostic pop
     if (Defined(Log_KEYS)) {
         uint8_t myPublicKeyHex[65];
         printHexPubKey(myPublicKeyHex, myPrivateKey);
diff -up ./dht/CJDHTConstants.h.warnings ./dht/CJDHTConstants.h
--- ./dht/CJDHTConstants.h.warnings	2017-02-21 17:56:22.000000000 -0500
+++ ./dht/CJDHTConstants.h	2017-02-24 12:36:55.762819179 -0500
@@ -17,7 +17,9 @@
 
 #include "benc/String.h"
 #include "util/version/Version.h"
-
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wpragmas"
+#pragma GCC diagnostic ignored "-Wunused-const-variable"
 // Signifying that this message is a query and defining the query type.
 static String* const CJDHTConstants_QUERY = String_CONST_SO("q");
 
@@ -61,5 +63,5 @@ static String* const CJDHTConstants_ENC_
 // Encoding scheme and index for the closest peer along the path.
 static String* const CJDHTConstants_PEER_ENC_SCHEME = String_CONST_SO("pes");
 static String* const CJDHTConstants_PEER_ENC_INDEX = String_CONST_SO("pei");
-
+#pragma GCC diagnostic pop
 #endif
diff -up ./dht/dhtcore/NodeStore.c.warnings ./dht/dhtcore/NodeStore.c
--- ./dht/dhtcore/NodeStore.c.warnings	2017-02-21 17:56:22.000000000 -0500
+++ ./dht/dhtcore/NodeStore.c	2017-02-24 12:44:19.822328184 -0500
@@ -1770,7 +1770,9 @@ struct NodeList* NodeStore_getPeers(uint
             break;
         }
         switch (j) {
-            default: Bits_memmove(out->nodes, &out->nodes[1], (j - 1) * sizeof(char*));
+            default:
+              Bits_memmove(out->nodes, &out->nodes[1], (j - 1) * sizeof(char*));
+              Gcc_FALLTHROUGH;
             case 1: out->nodes[j - 1] = next->child;
             case 0:;
         }
diff -up ./node_build/dependencies/libuv/test/test-getsockname.c.warnings ./node_build/dependencies/libuv/test/test-getsockname.c
--- ./node_build/dependencies/libuv/test/test-getsockname.c.warnings	2017-02-21 17:56:22.000000000 -0500
+++ ./node_build/dependencies/libuv/test/test-getsockname.c	2017-02-24 12:36:55.763819191 -0500
@@ -18,7 +18,7 @@
  * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  * IN THE SOFTWARE.
  */
-
+#pragma GCC diagnostic ignored "-Wstrict-aliasing"
 #include "uv.h"
 #include "task.h"
 
diff -up ./util/Gcc.h.warnings ./util/Gcc.h
--- ./util/Gcc.h.warnings	2017-02-24 12:41:36.926182554 -0500
+++ ./util/Gcc.h	2017-02-24 13:06:44.166644855 -0500
@@ -40,6 +40,11 @@
 #define Gcc_USE_RET \
     __attribute__ ((warn_unused_result))
 
+#if __GNUC__ >= 7
+#define Gcc_FALLTHROUGH \
+    __attribute__ ((fallthrough))
+#endif
+
 #elif defined(__clang__)
 
 #define Gcc_NORETURN \
@@ -50,6 +55,9 @@
 
 #endif
 
+#ifndef Gcc_FALLTHROUGH
+    #define Gcc_FALLTHROUGH (void)0
+#endif
 #ifndef Gcc_PRINTF
     #define Gcc_PRINTF( format_idx, arg_idx )
 #endif
diff -up ./util/platform/netdev/NetPlatform_linux.c.warnings ./util/platform/netdev/NetPlatform_linux.c
--- ./util/platform/netdev/NetPlatform_linux.c.warnings	2017-02-21 17:56:22.000000000 -0500
+++ ./util/platform/netdev/NetPlatform_linux.c	2017-02-24 12:36:55.763819191 -0500
@@ -165,7 +165,7 @@ void NetPlatform_addAddress(const char*
             Except_throw(eh, "ioctl(SIOCSIFADDR) failed: [%s]", strerror(err));
         }
 
-        uint32_t x = (uint32_t)~0 << (32 - prefixLen);
+        uint32_t x = ~0U << (32 - prefixLen);
         x = Endian_hostToBigEndian32(x);
         memcpy(&sin.sin_addr, &x, 4);
         memcpy(&ifRequest.ifr_addr, &sin, sizeof(struct sockaddr_in));
diff -up ./util/platform/Sockaddr.c.warnings ./util/platform/Sockaddr.c
--- ./util/platform/Sockaddr.c.warnings	2017-02-21 17:56:22.000000000 -0500
+++ ./util/platform/Sockaddr.c	2017-02-24 12:36:55.763819191 -0500
@@ -12,6 +12,7 @@
  * You should have received a copy of the GNU General Public License
  * along with this program.  If not, see <https://www.gnu.org/licenses/>.
  */
+#pragma GCC diagnostic ignored "-Wstrict-aliasing"
 #include "util/events/libuv/UvWrapper.h"
 #include "benc/String.h"
 #include "memory/Allocator.h"
