diff -up ./test/Main_fuzz_test.c.fuzz ./test/Main_fuzz_test.c
--- ./test/Main_fuzz_test.c.fuzz	2020-04-09 14:01:51.000000000 -0400
+++ ./test/Main_fuzz_test.c	2020-04-14 17:17:24.101260879 -0400
@@ -60,8 +60,8 @@ static Iface_DEFUN incomingTun(struct Me
 static void notLinkedYet(struct Context* ctx)
 {
     uint64_t now = Time_currentTimeMilliseconds(ctx->base);
-    if ((now - ctx->startTime) > 5000) {
-        Assert_failure("Failed to link in 5 seconds");
+    if ((now - ctx->startTime) > 200000) {
+        Assert_failure("Failed to link in 100 seconds");
     }
 }
 
diff -up ./util/test/Seccomp_test.c.fuzz ./util/test/Seccomp_test.c
--- ./util/test/Seccomp_test.c.fuzz	2020-04-15 20:21:02.563170274 -0400
+++ ./util/test/Seccomp_test.c	2020-04-15 20:21:22.893491368 -0400
@@ -70,7 +70,7 @@ static int child(char* pipeName, struct
     pipe->logger = logger;
     pipe->userData = alloc;
 
-    Timeout_setTimeout(timeout, eb, 2000, eb, alloc);
+    Timeout_setTimeout(timeout, eb, 3000, eb, alloc);
     EventBase_beginLoop(eb);
 
     return 0;
@@ -120,7 +120,7 @@ int main(int argc, char** argv)
 
     Assert_true(!Process_spawn(path, args, eb, alloc, NULL));
 
-    Timeout_setTimeout(timeout, NULL, 2000, eb, alloc);
+    Timeout_setTimeout(timeout, NULL, 3000, eb, alloc);
 
     EventBase_beginLoop(eb);
     return 0;
