diff -up ./client/cjdroute2.c.genconf ./client/cjdroute2.c
--- ./client/cjdroute2.c.genconf	2019-05-02 04:02:32.000000000 -0400
+++ ./client/cjdroute2.c	2019-05-08 12:14:53.146838389 -0400
@@ -336,7 +336,7 @@ static int genconf(struct Random* rand,
            "        // and ETHInterface will be unable to hot-add new interfaces\n"
            "        // Use { \"setuser\": 0 } to disable.\n"
            "        // Default: enabled with keepNetAdmin\n"
-           "        { \"setuser\": \"nobody\", \"keepNetAdmin\": 1 },\n"
+           "        { \"setuser\": \"cjdns\", \"keepNetAdmin\": 1 },\n"
            "\n"
            "        // Chroot changes the filesystem root directory which cjdns sees, blocking it\n"
            "        // from accessing files outside of the chroot sandbox, if the user does not\n"
@@ -348,7 +348,7 @@ static int genconf(struct Random* rand,
           }
           else {
     printf("        // Default: enabled (using \"/var/run\")\n"
-           "        { \"chroot\": \"/var/run/\" },\n");
+           "        { \"chroot\": \"/var/empty/cjdns\" },\n");
           }
     printf("\n"
            "        // Nofiles is a deprecated security feature which prevents cjdns from opening\n"
