diff -up ./node_build/builder.js.flagdup ./node_build/builder.js
--- ./node_build/builder.js.flagdup	2020-10-05 17:46:19.000000000 -0400
+++ ./node_build/builder.js	2023-05-02 15:01:37.727870212 -0400
@@ -182,7 +182,7 @@ const cc = function (
 ) {
     compiler(ctx, args, function (ret, out, err) {
         if (ret) {
-            return callback(error("gcc " + args.map(String).join(' ') + "\n\n" + err));
+            return callback(error(ctx.config.gcc + " " + args.map(String).join(' ') + "\n\n" + err));
         }
 
         if (err !== '') {
@@ -779,7 +779,7 @@ module.exports.configure = function (
     // if it returns undefined let's just assume 1
     // workaround, nodejs seems to be broken on openbsd (undefined result after second call)
     const cpus = Os.cpus();
-    const jobs = Math.floor((typeof cpus === 'undefined' ? 1 : cpus.length) * 1.25);
+    const jobs = 1;
 
     const pctx /*:Builder_PreCtx_t*/ = {
         buildStage: (_x,_y)=>{},
diff -up ./node_build/make.js.flagdup ./node_build/make.js
--- ./node_build/make.js.flagdup	2023-05-02 15:01:37.725870177 -0400
+++ ./node_build/make.js	2023-05-02 15:07:44.560178596 -0400
@@ -41,7 +41,6 @@ Builder.configure({
         '-std=c99',
         '-Wall',
         '-Wextra',
-        '-Werror',
         '-Wno-pointer-sign',
         '-Wmissing-prototypes',
         '-pedantic',
@@ -91,7 +90,7 @@ Builder.configure({
 
     if (!builder.config.crossCompiling) {
         if (NO_MARCH_FLAG.indexOf(process.arch) == -1) {
-            builder.config.cflags.push('-march=native');
+            builder.config.cflags.push('-mtune=generic');
         }
     }
 
@@ -144,14 +143,14 @@ Builder.configure({
     // with NEON on the BBB, or want to set -Os (OpenWrt)
     // Allow -O0 so while debugging all variables are present.
     if (CFLAGS) {
-        var cflags = CFLAGS.split(' ');
+        var cflags = CFLAGS.trim().split(/\s+/);
         cflags.forEach(function(flag) {
              if (/^\-O[^02s]$/.test(flag)) {
                 console.log("Skipping " + flag + ", assuming " + optimizeLevel + " instead.");
             } else if (/^\-O[02s]$/.test(flag)) {
                 optimizeLevel = flag;
             } else {
-                [].push.apply(builder.config.cflags, cflags);
+                builder.config.cflags.push(flag);
             }
         });
     }
@@ -164,7 +163,7 @@ Builder.configure({
     // We also need to pass various architecture/floating point flags to GCC when invoked as
     // a linker.
     if (LDFLAGS) {
-        [].push.apply(builder.config.ldflags, LDFLAGS.split(' '));
+        [].push.apply(builder.config.ldflags, LDFLAGS.trim().split(/\s+/));
     }
 
     if (android) {
