diff -up ./crypto/random/seed/SystemRandomSeed.c.sysctl ./crypto/random/seed/SystemRandomSeed.c
--- ./crypto/random/seed/SystemRandomSeed.c.sysctl	2019-08-06 12:28:23.084611733 -0400
+++ ./crypto/random/seed/SystemRandomSeed.c	2019-08-06 12:28:36.864803551 -0400
@@ -19,7 +19,6 @@
 #include "crypto/random/seed/RtlGenRandomSeed.h"
 #include "crypto/random/seed/BsdKernArndSysctlRandomSeed.h"
 #include "crypto/random/seed/DevUrandomRandomSeed.h"
-#include "crypto/random/seed/LinuxRandomUuidSysctlRandomSeed.h"
 #include "crypto/random/seed/ProcSysKernelRandomUuidRandomSeed.h"
 #include "crypto/random/seed/GetEntropyRandomSeed.h"
 #include "crypto/random/seed/SystemRandomSeed.h"
