diff -up ./memory/Allocator.c.32bit ./memory/Allocator.c
--- ./memory/Allocator.c.32bit	2020-09-29 14:10:32.115169486 -0400
+++ ./memory/Allocator.c	2020-09-29 14:16:54.682448839 -0400
@@ -59,8 +59,8 @@ static void unroll(struct Allocator_pvt*
         writeUnroller(&childUnroller);
         fprintf(stderr, "%s:%ld [%lu] bytes at [0x%lx]\n",
                 allocation->fileName,
-                allocation->lineNum,
-                allocation->pub.size,
+                (long)allocation->lineNum,
+                (unsigned long)allocation->pub.size,
                 (long)(uintptr_t)allocation);
         allocation = allocation->next;
     }
