diff -up ./node_build/dependencies/libuv/gyp_uv.py.python3 ./node_build/dependencies/libuv/gyp_uv.py
--- ./node_build/dependencies/libuv/gyp_uv.py.python3	2019-08-06 18:58:48.000000000 -0400
+++ ./node_build/dependencies/libuv/gyp_uv.py	2019-09-19 13:11:51.125240959 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 
 import os
 import platform
@@ -29,7 +29,7 @@ def host_arch():
 def run_gyp(args):
   rc = gyp.main(args)
   if rc != 0:
-    print 'Error running GYP'
+    print('Error running GYP')
     sys.exit(rc)
 
 
@@ -80,5 +80,5 @@ if __name__ == '__main__':
     args.append('-Dcomponent=static_library')
 
   gyp_args = list(args)
-  print gyp_args
+  print(gyp_args)
   run_gyp(gyp_args)
