diff -up ./node_modules/ronn/bin/ronn.js.puts ./node_modules/ronn/bin/ronn.js
--- ./node_modules/ronn/bin/ronn.js.puts	2019-08-03 18:32:03.560415945 -0400
+++ ./node_modules/ronn/bin/ronn.js	2019-08-03 18:33:18.600372936 -0400
@@ -2,7 +2,8 @@
 
 var RONN_VERSION = "0.4.0";
 
-var sys = require('util');
+var console = require('console');
+var sys = { 'puts': console.log, 'debug': console.error };
 var fs = require('fs');
 var path = require('path');
 var opts = require('opts');
diff -up ./node_modules/ronn/lib/ronn.js.puts ./node_modules/ronn/lib/ronn.js
--- ./node_modules/ronn/lib/ronn.js.puts	2019-08-03 18:32:11.703523935 -0400
+++ ./node_modules/ronn/lib/ronn.js	2019-08-03 18:33:02.578184733 -0400
@@ -4,7 +4,8 @@
  */
 
 var md = require('markdown').markdown;
-var sys = require('util');
+var console = require('console');
+var sys = { 'puts': console.log, 'debug': console.error };
 
 /* exports Ronn class
  * usage :
